### ISSUE TITLE
Suggestion: Change CursorShape in Code

### ISSUE THREAD

**Issue by Phoenix1747** [godotengine/godot#18043]
**Godot version:**

3.0.2-stable official

**OS/device including version:**

Arch Linux

**Feature description:**

It would be really helpful if there was a function or any other way one could change the mouse cursor shape in the code itself (GDScript), instead of relying solely on the Control-inherited nodes to change it. I couldn't find any way to do so, therefore, here is my suggestion :)

**Comment by Phoenix1747** [comment]
Yeah, I think it would be nice if you could set the different custom mouse cursors with `Input.set_custom_mous_cursor()` and then actually programmatically change the shape to one of the CursorShape enums that I set beforehand with the function.

**Comment by RabbitB** [comment]
I definitely don't understand the question then, because `set_custom_mouse_cursor` allows you to define what cursor type you want to change as well as the hot-spot for each cursor.

Are we talking about an animated cursor?

What I'm not understanding is exactly what functionality you want, that isn't covered by the method, because beyond animation, I can't envision anything you could do with a cursor, that you can't do through that method.

```python
void set_custom_mouse_cursor ( Resource image, int shape=0, Vector2 hotspot=Vector2( 0, 0 ) )
    Set a custom mouse cursor image, which is only visible inside the game window.
    The hotspot can also be specified. See enum CURSOR_* for the list of shapes.
```

**Comment by Phoenix1747** [comment]
I'm not talking about animation, so let me explain a little more what exactly I am trying to do.

You can set different custom mouse cursors with `void set_custom_mouse_cursor ( Resource image, int shape=0, Vector2 hotspot=Vector2( 0, 0 ) )`. I can set any of the `enum CursorShape` to a custom cursor by using this, correct?

Now, the only way to change the mouse cursor (atm) is by setting the `CursorShape mouse_default_cursor_shape` to any of the `enum CursorShape` I customized earlier. This however only works in nodes that inherit from `Control` and only inside their rect size.

So the only way to change the cursor globally is to set the MouseShape 0 (aka "Arrow") to a custom cursor to get it for the whole game window constantly. Now it would be very helpful if there was a function in `Input` that changed the mouse cursor shape to any of the previously customized ones with a function like for example `void set_mouse_cursor_shape ( enum CursorShape )`. A somehow comparable function that already exists in `Input` is `void set_mouse_mode ( int mode )`.

<hr>

tl;dr Set all CursorShapes with the function `void set_custom_mouse_cursor ( Resource image, int shape=0, Vector2 hotspot=Vector2( 0, 0 ) )` like so:
```python
set_custom_mouse_cursor(image1, 0)
set_custom_mouse_cursor(image2, 1)
...
```

and then change to any custom mouse cursor shape like so:
```python
set_mouse_cursor_shape(1)
```
...to get the customized mouse cursor for the whole game window.

**Comment by guilhermefelipecgs** [comment]
@Phoenix1747 Why you can't just use ```set_custom_mouse_cursor```? if this method isn't having some performance problem I don't think it's worh it to implement ```set_mouse_cursor_shape``` IMO.

And the workaround seems fine to me if it's working.

**Comment by Phoenix1747** [comment]
Well, having to set a control node full rect only for the sake of cursors doesn't seem clean at all. Also there might be issues with overlapping nodes that change the cursor again, but that is too much to go into for now - this really depends on the project use case.

Furthermore, it seems that the basic functionality for `set_mouse_cursor_shape` is already there with any `Control`-inherited node being able to change it automatically, so why not like opening up the function for the developer to use in GDScript. This doesn't seem like a special corner case too, if you can customize the different cursor shapes, why not also be able to use these exact shapes in GDScript?

**Comment by guilhermefelipecgs** [comment]
@Phoenix1747 I made a pr #18093 to implement this method ```Input.set_default_cursor_shape``` on gdscript, is this what you want?
